File: libs/c3dclassessdk/ccore/cconstructs/chook/CHook.js
//-------------------------------------------------------------------------------------------------------
// file: chook.php
// desc: provides a way to store php handler function that will be executed later by the ckernal object
//-------------------------------------------------------------------------------------------------------

//-----------------------------------------------------------
// name: CHook
// desc: singleton object used to store handler functions
//-----------------------------------------------------------
var CHook = new Class ({
	ClassMethods : {		
	m_chashhook : null,
	add : function( strhookname, fnhandler ){
		if( strhookname == "" || strhookname == null || fnhandler == null )
			return false;
		if( CHook.m_chashhook == null )
			CHook.m_chashhook = new CHash();		
		if( CHook.m_chashhook.containsKey( strhookname ) == false )
			CHook.m_chashhook.set( strhookname, new Array() );
		var chook = CHook.m_chashhook.get( strhookname )
		if( chook ) 
			chook.push( fnhandler );
		return true;
	}, // end add()
	remove : function( strhookname, fnhandler ){
		if( strhookname == "" || strhookname == null || CHook.m_chashhook == null )
			return false;
		if( fnhandler == null )
			CHook.m_chashhook.remove( strhookname );
		else if( CHook.m_chashhook.containsKey( strhookname ) == true )
			CHook.m_chashhook.get( strhookname ).remove( fnhandler );
		return true;
	}, // end remove()
	fire : function( strhookname ){	 
		var chook = null;
		if( CHook.m_chashhook == null || ( chook = CHook.m_chashhook.get( strhookname ) ) == null )
			return false;		
		chook.visit( CHook.doCallback ); 
		return true;	
	}, // end fire()
	doCallback : function ( key, fnhandler, params ){
		if( isFunction( fnhandler ) ) fnhandler( key, params ); 
	} // end doCallback()
	} // end ClassMethods
}); // end CHook

File: libs/c3dclassessdk/ccore/cconstructs/chook/CHook.java
//-------------------------------------------------------------------------------------------------------
// file: CHook.java
// desc: provides a way to store php handler function that will be executed later by the ckernal object
//-------------------------------------------------------------------------------------------------------
package c3dclasses;

//-----------------------------------------------------------
// name: CHook
// desc: singleton object used to store handler functions
//-----------------------------------------------------------
public class CHook {
	public static CHash m_chashhook = null;
	public static boolean add(String strhookname, CFunction fnhandler) {	
		if(strhookname == "" || strhookname == null || fnhandler == null)
			return false;
		if(CHook.m_chashhook == null)
			CHook.m_chashhook = _.chash();		
		if(CHook.m_chashhook.containsKey(strhookname) == false)
			CHook.m_chashhook.set(strhookname, _.carray());
		CArray chook = (CArray) CHook.m_chashhook.get(strhookname);
		if(chook != null) 
			chook.push(fnhandler);
		return true;
	} // end add()
	
	public static boolean remove(String strhookname, CFunction fnhandler) {
		if(strhookname == "" || strhookname == null || CHook.m_chashhook == null)
			return false;
		if(fnhandler == null)
			CHook.m_chashhook.remove(strhookname);
		else if(CHook.m_chashhook.containsKey(strhookname) == true) {
			CArray carray = (CArray) CHook.m_chashhook.get(strhookname);
			if(carray != null) 
				carray.remove(fnhandler);
		} // end else if
		return true;
	} // end remove()
	
	public static boolean fire(String strhookname) {	 
		if(CHook.m_chashhook == null)
			return false;
		CArray chook = (CArray) CHook.m_chashhook.get(strhookname);
		if(chook == null)
			return false;
		for(int i=0; i<chook.length(); i++)
			chook._cfunction(i).call((CArray)null);
		return true;	
	} // end fire()
} // end CHook